File: doconnect_frontend_final/src/services/questionService.js
/** 

 * Question Service 

 * All question-related API calls 

 */ 

import axiosInstance from '../api/axiosConfig'; 

 

const questionService = { 

 

  getAllQuestions: async () => { 

    const res = await axiosInstance 

                          .get('/questions'); 

    return res.data; 

  }, 

 

  getQuestionById: async (id) => { 

    const res = await axiosInstance 

                      .get(`/questions/${id}`); 

    return res.data; 

  }, 

 

  createQuestion: async (data) => { 

    const res = await axiosInstance 

                  .post('/questions', data); 

    return res.data; 

  }, 

 

  updateQuestion: async (id, data) => { 

    const res = await axiosInstance 

                .put(`/questions/${id}`, data); 

    return res.data; 

  }, 

 

  deleteQuestion: async (id) => { 

    const res = await axiosInstance 

                  .delete(`/questions/${id}`); 

    return res.data; 

  }, 

 

  searchQuestions: async (keyword) => { 

    const res = await axiosInstance 

        .get(`/questions/search?keyword=${keyword}`); 

    return res.data; 

  }, 

 

  getTrending: async () => { 

    const res = await axiosInstance 

                .get('/questions/trending'); 

    return res.data; 

  }, 

 

  getByTag: async (tagName) => { 

    const res = await axiosInstance 

            .get(`/questions/tag/${tagName}`); 

    return res.data; 

  }, 

 

  voteQuestion: async (id, type) => { 

    const res = await axiosInstance 

        .put(`/questions/${id}/vote?type=${type}`); 

    return res.data; 

  }, 

}; 

 

export default questionService; 

File: DoConnect_Final/src/main/java/com/doconnect/service/interfaces/QuestionService.java
package com.doconnect.service.interfaces;

import java.util.List;

import com.doconnect.dto.question.QuestionRequestDTO;
import com.doconnect.dto.question.QuestionResponseDTO;

public interface QuestionService { 

    QuestionResponseDTO createQuestion( 

            QuestionRequestDTO dto, 

            String email); 

    List<QuestionResponseDTO> getAllQuestions(); 

    QuestionResponseDTO getQuestionById(Long id); 

    QuestionResponseDTO updateQuestion( 

            Long id, 

            QuestionRequestDTO dto, 

            String email); 

    void deleteQuestion(Long id, String email); 

    List<QuestionResponseDTO> searchQuestions( 

            String keyword); 

    List<QuestionResponseDTO> getTrendingQuestions(); 

    List<QuestionResponseDTO> getQuestionsByTag( 

            String tagName); 

    QuestionResponseDTO voteQuestion( 

            Long id, String type); 

} 